packages/react-components/src/components/ImageGallery.tsx

```diff
@@ -0,0 +1,171 @@
+// Licensed under the MIT license.
+
+import {
+  DefaultButton,
+  IModalStyleProps,
+  IModalStyles,
+  IOverlayStyleProps,
+  IOverlayStyles,
+  IStyle,
+  IStyleFunctionOrObject,
+  Icon,
+  IconButton,
+  Layer,
+  Modal,
+  Stack,
+  mergeStyles
+} from '@fluentui/react';
+
+import React from 'react';
+import { BaseCustomStyles } from '../types';
+import {
+  cancelIcon,
+  closeButtonStyles,
+  controlBarContainerStyle,
+  downloadButtonStyle,
+  downloadIcon,
+  downloadIconStyle,
+  focusTrapZoneStyle,
+  headerStyle,
+  imageContainer,
+  imageStyle,
+  overlayStyles,
+  scrollableContentStyle,
+  smallDownloadButtonContainerStyle,
+  titleBarContainerStyle,
+  titleStyle
+} from './styles/ImageGallery.style';
+import { useTheme } from '../theming/FluentThemeProvider';
+import { isDarkThemed } from '../theming/themeUtils';
+
+/**
+ * Fluent styles for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryStylesProps extends BaseCustomStyles {
+  /** Styles for the ImageGallery Modal. */
+  modal?: IStyleFunctionOrObject<IModalStyleProps, IModalStyles>;
+  /** Styles for the ImageGallery Modal overlay. */
+  overlay?: IStyleFunctionOrObject<IOverlayStyleProps, IOverlayStyles>;
+  /** Styles for the ImageGallery header bar. */
+  header?: IStyle;
+  /** Styles for the ImageGallery titleBar container. */
+  titleBarContainer?: IStyle;
+  /** styles for the title label */
+  title?: IStyle;
+  /** Styles for the ImageGallery controlBar container. */
+  controlBarContainer?: IStyle;
+  /** Styles for the download button. */
+  downloadButton?: IStyle;
+  /** Styles for the small download button when screen width is smaller than 25 rem. */
+  smallDownloadButton?: IStyle;
+  /** Styles for the close modal icon. */
+  closeIcon?: IStyle;
+  /** Styles for the image. */
+  image?: IStyle;
+}
+
+/**
+ * Props for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryImageProps {
+  imageUrl: string;
+  downloadFileName: string;
+  altText?: string;
+  title?: string;
+  titleIcon?: JSX.Element;
+}
+
+/**
+ * Props for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryProps {
+  images: Array<ImageGalleryImageProps>;
+  onDismiss: () => void;
+  onImageDownloadButtonClicked: (imageUrl: string, downloadFileName: string) => void;
+  modalLayerHostId?: string;
+  /**
+   * Allows users to pass in an object contains custom CSS styles.
+   * @Example
+   * ```
+   * <ImageGallery styles={{ image: { background: 'blue' } }} />
+   * ```
+   */
+  styles?: ImageGalleryStylesProps;
+}
+
+/**
+ * @beta
+ */
+export const ImageGallery = (props: ImageGalleryProps): JSX.Element => {
+  const { images, modalLayerHostId, onImageDownloadButtonClicked, onDismiss, styles } = props;
+  const theme = useTheme();
+  const isDarkTheme = isDarkThemed(theme);
+
+  const downloadButtonTitleString = 'Download';
+  const closeString = 'Close';
+  const defaultAltText = 'image';
+
+  if (images.length <= 0) {
```

Comment:
Does it make sense to throw an error in this case?

Reply:
> Does it make sense to throw an error in this case?

It made sense to return an empty div when there's no images in the array. However, now that we switched to images.length <= startIndex, seems like we should find a way to throw an error to Contoso developer but not breaking the UI for end user?